librz/core/analysis_tp.c

```diff
@@ -9,14 +9,17 @@
 #include <rz_core.h>
 #define LOOP_MAX 10
 
-static bool analysis_emul_init(RzCore *core, RzConfigHold *hc, RzDebugTrace **dt, RzAnalysisEsilTrace **et) {
-	if (!core->analysis->esil) {
+static bool analysis_emul_init(RzCore *core, RzConfigHold *hc, RzDebugTrace **dt, RzAnalysisEsilTrace **et, RzAnalysisRzilTrace **rt) {
+	if (!core->analysis->esil || !core->analysis->rzil) {
 		return false;
 	}
 	*dt = core->dbg->trace;
 	*et = core->analysis->esil->trace;
+
 	core->dbg->trace = rz_debug_trace_new();
 	core->analysis->esil->trace = rz_analysis_esil_trace_new(core->analysis->esil);
+	core->analysis->rzil->trace = rz_analysis_rzil_trace_new(core->analysis, core->analysis->rzil);
```

Comment:
if you have `core->analysis` why passing `core->analysis->rzil` ?

Reply:
Yes it passes these 2 because I followed the ESIL api. We can only maintain one argument `core->analysis`.

Reply:
Yes, lets skip and use just `core->analysis` for both.